tasks/eth/base-002-fp-upgrade/EXEC.md

```diff
@@ -0,0 +1,127 @@
+# FP Upgrade - Base Mainnet
+Upgrades the `OptimismPortal` and `SystemConfig` implementations
+
+The batch will be executed on chain ID `1`, and contains `7` transactions.
+
+## Tx #1: Upgrade OptimismPortal to StorageSetter
+Upgrade OptimismPortal to StorageSetter and reset `initializing`
+
+**Function Signature:** `upgradeAndCall(address,address,bytes)`
+
+**To:** `0x0475cBCAebd9CE8AfA5025828d5b98DFb67E059E`
+
+**Value:** `0 WEI`
+
+**Raw Input Data:** `0x9623609d00000000000000000000000049048044d57e1c92a77f79988d21fa8faf74e97e000000000000000000000000d81f43edbcacb4c29a9ba38a13ee5d79278270cc000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000444e91db080000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000`
+
+### Inputs
+**_implementation:** `0xd81f43eDBCAcb4c29a9bA38a13Ee5d79278270cC`
+
+**_proxy:** `0x49048044D57e1C92A77f79988d21Fa8fAF74E97e`
+
+**_data:** `0x4e91db0800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000`
+
+
+## Tx #2: Reset l2Sender in OptimismPortalProxy
+Pre-initialization of the OptimismPortal2
+
+**Function Signature:** `setAddress(bytes32,address)`
+
+**To:** `0x49048044D57e1C92A77f79988d21Fa8fAF74E97e`
+
+**Value:** `0 WEI`
+
+**Raw Input Data:** `0xca446dd900000000000000000000000000000000000000000000000000000000000000320000000000000000000000000000000000000000000000000000000000000000`
+
+### Inputs
+**_slot:** `0x0000000000000000000000000000000000000000000000000000000000000032`
+
+**_address:** `0x0000000000000000000000000000000000000000`
+
+
+## Tx #3: Upgrade the OptimismPortal
+Upgrade and initialize the OptimismPortal to OptimismPortal2 (3.10.0)
+
+**Function Signature:** `upgradeAndCall(address,address,bytes)`
+
+**To:** `0x0475cBCAebd9CE8AfA5025828d5b98DFb67E059E`
+
+**Value:** `0 WEI`
+
+**Raw Input Data:** `0x9623609d00000000000000000000000049048044d57e1c92a77f79988d21fa8faf74e97e00000000000000000000000049048044d57e1c92a77f79988d21fa8faf74e97e00000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000014e2f826324b2faf99e513d16d266c3f80ae87832b000000000000000000000000`
+
+### Inputs
+**_data:** `0x8e819e5400000000000000000000000043edb88c4b80fdd2adff2412a7bebf9df42cb40e00000`
```

Comment:
Yep @Ethnical just noticed this as well, also wondering this. We should be using the full calldata for clarity. It's suspicious to me that this doesn't revert in solidity for the calldata being too short...

Reply:
We see the same issue seems that there is an issue with a copy paste because the notion doc https://www.notion.so/oplabs/Preparing-Base-Mainnet-s-Permissionless-FP-Upgrade-126f153ee162803e9a17e70895953c3c?pvs=4
Doesn't show the same length and content for this calldata.

Reply:
Good catch, we just ran into that ourselves and we're debugging it now. This is how it should be generated:

```
cast calldata 'initialize(address,address,address,uint32)' \
0x43edB88C4B80fDD2AdFF2412A7BebF9dF42cB40e \
0x73a79Fab69143498Ed3712e519A88a918e1f4072 \
0x95703e0982140D16f8ebA6d158FccEde42f04a4C \
0

0x8e819e5400000000000000000000000043edb88c4b80fdd2adff2412a7bebf9df42cb40e00000000000000000000000073a79fab69143498ed3712e519a88a918e1f407200000000000000000000000095703e0982140d16f8eba6d158fccede42f04a4c0000000000000000000000000000000000000000000000000000000000000000
```

Reply:
In the tenderly trace, I do see the correct data being called, so something does seem to mismatch between the `input.json` payload (which also has the same too-short-calldata) and what's shown in the sim
<img width="1933" alt="image" src="https://github.com/user-attachments/assets/d3a53d94-4709-4ebb-920a-64c3cd8a7897">

```
$ cast calldata-decode "initialize(address,address,address,uint32)" 0x8e819e5400000000000000000000000043edb88c4b80fdd2adff2412a7bebf9df42cb40e00000000000000000000000073a79fab69143498ed3712e519a88a918e1f407200000000000000000000000095703e0982140d16f8eba6d158fccede42f04a4c0000000000000000000000000000000000000000000000000000000000000000

0x43edB88C4B80fDD2AdFF2412A7BebF9dF42cB40e
0x73a79Fab69143498Ed3712e519A88a918e1f4072
0x95703e0982140D16f8ebA6d158FccEde42f04a4C
0
```

Reply:
Ok, the `input.json` and `EXEC.md` were manually edited from OP Mainnet's, resulting in this inconsistency. We are going to regenerate them all using [`msup`](https://github.com/clabby/msup/tree/main)  which is how the OP Mainnet one was generated

Reply:
Actually, `msup` is not great UX, so we are going to:
- Delete `EXEC.md` since this is optional
- Review the state diff to make sure everything is correct. If it is, this means the actual data in `input.json` is correct
- Manually review the `input.json` to ensure `contractInputsValues` to ensure they are consistent

Reply:
These reviews have been completed